### PR TITLE
fix: `NoSpacesAfterFunctionNameFixer` - do not remove space if the opening parenthesis part of an expression

### PR DESCRIPTION
**Pull request by kubawerlos**
Fixes https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/5950

### CODE REVIEW

File: tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php [hunk 1]
```diff
@@ -167,6 +167,21 @@ function TisMy    ($p1)
             '<?php $a()(1);',
             '<?php $a () (1);',
         ];
+
+        yield [
```
Comment (mvorisek) [inline tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php:171]: https://3v4l.org/CuJvh should be tested as well

not sure about other expressions like `throw ();`

Comment (mvorisek) [reply tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php:171]: LGTM

File: tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php
Comment (Wirone) [inline tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php:1]: I believe that yields' keys would make test cases more descriptive and easier to grasp.

**Review by Wirone — APPROVED** [review]
Can be merged as-is, comment is non-blocking.

File: tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php
Comment (kubawerlos) [reply tests/Fixer/FunctionNotation/NoSpacesAfterFunctionNameFixerTest.php:1]: Sometimes, sometimes you want to have a different example of the same issue and there is dilemma of putting it in one code chunk or to split into a few smaller code samples.

**Pull request merged**